mpire/comms.py

```diff
@@ -410,11 +411,26 @@ def get_exception(self) -> Tuple[type, str]:
         with DelayedKeyboardInterrupt():
             return self._exception_queue.get(block=True)
 
-    def task_done_exception(self) -> None:
+    def task_done_exception(self, progress_bar: bool = False) -> None:
         """
         Signal that we've completed a task for the exception queue
+
+        :param progress_bar: Whether the exception was handled for the progress bar
         """
         self._exception_queue.task_done()
+        self._exception_received[0].set() if not progress_bar else self._exception_received[1].set()
```

Comment:
If you keep _exception_received as a tuple:

```
self._exception_received[postgres_bar].set()
```

For fanciness points! :P

Reply:
Although it work in this case, I'm generally not a big fan of using a boolean as an integer. But I might change the tuple to two separate variables.